cmake/kokkos_enable_options.cmake

```diff
@@ -86,6 +86,8 @@ mark_as_advanced(Kokkos_ENABLE_IMPL_MDSPAN)
 mark_as_advanced(Kokkos_ENABLE_MDSPAN_EXTERNAL)
 mark_as_advanced(Kokkos_ENABLE_IMPL_SKIP_COMPILER_MDSPAN)
 
+KOKKOS_ENABLE_OPTION(NO_EXCEPTIONS ON BOOL "Whether to ban the use of exceptions")
```

Comment:
Planing to mark it as advanced.  Let me know if you disagree.

Reply:
(And will default to OFF obviously)

Reply:
Good idea to mark it as advanced.

Reply:
I find `NO_EXCEPTIONS=OFF/ON` harder to read than `USE_EXCEPTIONS=ON/OFF` but I can also understand this choice since the flag is `-fno-exceptions` for a lot of compilers.